test/libyul/yulOptimizerTests/loadResolver/merge_unknown_write.yul

```diff
@@ -0,0 +1,20 @@
+{
+    mstore(calldataload(0), calldataload(10))
+    if calldataload(1) {
+        mstore(0, 1)
+    }
+    let t := mload(0)
+    let q := mload(calldataload(0))
+    sstore(t, q)
+}
+// ====
+// step: loadResolver
+// ----
+// {
+//     let _2 := calldataload(10)
+//     let _3 := 0
+//     mstore(calldataload(_3), _2)
+//     let _5 := 1
+//     if calldataload(_5) { mstore(_3, _5) }
+//     sstore(mload(_3), _2)
```

Comment:
This looks faulty to me. The second argument to `sstore` is `mload(calldataload(0))` - but now it is replaced by the value stored there in line 2. We might have overwritten that value in line 4, though, so it has to be re-loaded...